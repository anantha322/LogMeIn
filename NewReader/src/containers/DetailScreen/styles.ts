import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  highlight: {
    fontWeight: '700',
  },
  overviewText: {
    marginBottom: 10,
    padding: 15,
  },
  container: {
    margin: 0,
    padding: 0,
  },
  headerContainer: {
    flexDirection: 'row',
    padding: 15,
  },
  headerTitle: {
    flex: 1,
  },
});
