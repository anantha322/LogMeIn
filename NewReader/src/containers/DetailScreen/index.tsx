import React from 'react';
import {Linking, Text, View} from 'react-native';
import {Card, Icon} from 'react-native-elements';
import {styles} from './styles';
import {RouteProp} from '@react-navigation/native';
import {StackNavigationProp} from '@react-navigation/stack';
import {NewsListItem} from '../../components';

type RootStackParamList = {
  Home: undefined;
  Details: {selectedNewsFeed: any};
};

type ProfileScreenRouteProp = RouteProp<RootStackParamList, 'Details'>;

type ProfileScreenNavigationProp = StackNavigationProp<
  RootStackParamList,
  'Details'
>;

type Props = {
  route: ProfileScreenRouteProp;
  navigation: ProfileScreenNavigationProp;
};

const DetailScreen = ({route, navigation}: Props) => {
  const {selectedNewsFeed} = route.params;
  const renderHeader = () => {
    return (
      <View style={styles.headerContainer}>
        <View style={styles.headerTitle}>
          <Card.Title>{selectedNewsFeed.title}</Card.Title>
        </View>
        <Icon onPress={navigation.goBack} name="close" color="black" />
      </View>
    );
  };
  const onPressReadMore = (url: string) => {
    Linking.canOpenURL(url).then(supported => {
      if (supported) {
        Linking.openURL(url);
      } else {
        console.log("Don't know how to open URI: " + url);
      }
    });
  };
  return (
    <Card containerStyle={styles.container}>
      {renderHeader()}
      {selectedNewsFeed.description && (
        <Text style={styles.overviewText}>{selectedNewsFeed.description}</Text>
      )}
      {selectedNewsFeed && selectedNewsFeed?.id && (
        <NewsListItem
          id={selectedNewsFeed.id}
          title={'Read More ...'}
          onPressItem={() => onPressReadMore(selectedNewsFeed.id)}
        />
      )}
    </Card>
  );
};

export default DetailScreen;
