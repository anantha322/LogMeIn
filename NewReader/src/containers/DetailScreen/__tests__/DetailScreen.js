import React from 'react';
import {Text} from 'react-native';
import {Icon, Card} from 'react-native-elements';
import renderer from 'react-test-renderer';
import DetailScreen from '..';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({adapter: new Adapter()});

const initialState = {};
const mockStore = configureStore();
let store;
const selectedNewsFeed = {
  id: 123,
  onPressItem: item => {
    console.log(item);
  },
  title: 'Test',
  description: 'Test Test test',
};
const props = {
  route: {
    params: {
      selectedNewsFeed: selectedNewsFeed,
    },
  },
  navigation: {
    goBack: jest.fn(),
  },
};
describe('render Detail screen', () => {
  const wrapper = Enzyme.shallow(<DetailScreen {...props} />);
  test('renders correctly', () => {
    store = mockStore(initialState);

    const tree = renderer
      .create(
        <Provider store={store}>
          <DetailScreen {...props} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('close button render and called on press', () => {
    const buttonProps = wrapper.find(Icon).first().props();
    buttonProps.onPress();
    expect(buttonProps.name).toEqual('close');
    expect(props.navigation.goBack).toBeCalled();
  });

  test('should have the title passed from props', () => {
    expect(wrapper.find(Card.Title).first().props().children).toEqual(
      props.route.params.selectedNewsFeed.title,
    );
  });

  test('should have the description passed from props', () => {
    expect(wrapper.find(Text).first().props().children).toEqual(
      props.route.params.selectedNewsFeed.description,
    );
  });
});
