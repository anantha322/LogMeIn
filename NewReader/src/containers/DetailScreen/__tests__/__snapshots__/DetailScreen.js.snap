// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Detail screen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#e1e8ee",
      "borderWidth": 1,
      "margin": 0,
      "marginBottom": 0,
      "padding": 0,
      "shadowColor": "rgba(0,0,0, .2)",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "padding": 15,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#43484d",
              "fontSize": 18,
              "fontWeight": "bold",
              "marginBottom": 15,
              "textAlign": "center",
            }
          }
          testID="cardTitle"
        >
          Test
        </Text>
      </View>
      <View
        style={
          Object {
            "overflow": "hidden",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "black",
                    "fontSize": 24,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              Óóç
            </Text>
          </View>
        </View>
      </View>
    </View>
    <Text
      style={
        Object {
          "marginBottom": 10,
          "padding": 15,
        }
      }
    >
      Test Test test
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderBottomWidth": 0.5,
            "borderColor": "#bcbbc1",
            "flexDirection": "row",
            "padding": 14,
          }
        }
        testID="padView"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "flex-start",
                "flex": 1,
                "justifyContent": "center",
              },
              undefined,
            ]
          }
          theme={
            Object {
              "colors": Object {
                "black": "#242424",
                "disabled": "hsl(208, 8%, 90%)",
                "divider": "#bcbbc1",
                "error": "#ff190c",
                "grey0": "#393e42",
                "grey1": "#43484d",
                "grey2": "#5e6977",
                "grey3": "#86939e",
                "grey4": "#bdc6cf",
                "grey5": "#e1e8ee",
                "greyOutline": "#bbb",
                "platform": Object {
                  "android": Object {
                    "error": "#f44336",
                    "grey": "rgba(0, 0, 0, 0.54)",
                    "primary": "#2196f3",
                    "searchBg": "#dcdce1",
                    "secondary": "#9C27B0",
                    "success": "#4caf50",
                    "warning": "#ffeb3b",
                  },
                  "default": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "ios": Object {
                    "error": "#ff3b30",
                    "grey": "#7d7d7d",
                    "primary": "#007aff",
                    "searchBg": "#dcdce1",
                    "secondary": "#5856d6",
                    "success": "#4cd964",
                    "warning": "#ffcc00",
                  },
                  "web": Object {
                    "error": "#ff190c",
                    "grey": "#393e42",
                    "primary": "#2089dc",
                    "searchBg": "#303337",
                    "secondary": "#ca71eb",
                    "success": "#52c41a",
                    "warning": "#faad14",
                  },
                },
                "primary": "#2089dc",
                "searchBg": "#303337",
                "secondary": "#ca71eb",
                "success": "#52c41a",
                "warning": "#faad14",
                "white": "#ffffff",
              },
            }
          }
        >
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#242424",
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
            testID="listItemTitle"
          >
            Read More ...
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
